samcli/local/docker/container.py

```diff
@@ -272,6 +274,31 @@ def start(self, input_data=None):
         # Start the container
         real_container.start()
 
+        # Wait for port to be open
+        self.wait_for_port()
+
+    def wait_for_port(self):
+        """
+        Waits until the host machine port that Docker binds to is open.
+        """
+        start_time = time.time()
+        sleep = 0.1
+        timeout = 10
```

Comment:
for my tests it's only been taking a couple of seconds at most. Is this a good number for the timeout? Should it be configurable by the customer? (seems very specific though)

Reply:
>For some context, before adding this new function to wait for the port, a local invoke would time out after either exhausting the retries (3) or the function timeout (default is 3 seconds), whichever comes first (meaning it wouldn't wait for the port for more than 3 seconds, or the function timeout, at most).
This new one would be another timeout, before those two start.

Reply:
> so do you mean SAM should not wait more than 3 seconds ?
I think it should wait at least like 5 seconds anyway, and making it configurable through an environment variable seems like a good idea 👍